packages/clerk-js/src/ui/UserButton/UserButton.tsx

```diff
@@ -10,8 +10,9 @@ import { UserButtonPopover } from './UserButtonPopover';
 import { UserButtonTrigger } from './UserButtonTrigger';
 
 const _UserButton = () => {
+  const { isOpen: isButtonOpenFromOutside } = useUserButtonContext();
   const { floating, reference, styles, toggle, isOpen } = usePopover({
-    defaultOpen: false,
+    defaultOpen: isButtonOpenFromOutside,
```

Comment:
⚠️ The `defaultOpen` flag is only used for the **first** component mount, so changing the `isButtonOpenFromOutside` prop will not affect the behavior of the hook

Reply:
@nikosdouvlis `defaultOpen` does change since the usePopover gets rerenderd. That is how we setState within it